app/src/main/kotlin/com/hippo/unifile/TreeDocumentFile.kt

```diff
@@ -23,10 +23,21 @@ import splitties.init.appCtx
 
 class TreeDocumentFile(
     parent: TreeDocumentFile?,
-    override val uri: Uri,
-    name: String? = null,
-    mimeType: String? = null,
+    private var backingUri: Uri = Uri.EMPTY,
+    private val givenName: String? = null,
+    private val givenMimeType: String? = null,
 ) : CachingFile<TreeDocumentFile>(parent) {
+
+    override val uri: Uri
+        get() = prepareUri().let { backingUri }
+
+    private fun prepareUri() {
+        if (backingUri != Uri.EMPTY) return
+        val parent = parent ?: return
+        val f = parent.findFile(name) ?: return
+        backingUri = f.uri
```

Comment:
This does not synergize with the caching mechanism.
The cached file object is different from this, so when the cache won't be updated if this got deleted.

Reply:
Let's evict cache by file name?

Reply:
Or we just replace cache with a Map<String, Unifile> and update it?

Reply:
Another problem is `resolve` always constructs an uncached new object instead of reusing the cached ones, so children of the same file will be looked up multiple times.

Reply:
Done

Reply:
Actually, maybe we do want `resolve`/`findFile` to return an uncached new object so the parent only holds cache for its children not descendants to avoid memory explosion.

Reply:
Maybe we can be more aggressive, the cache value is just Uri or String(name)

Reply:
Or we should just do cache on SpiderDen side, I don't see any necessity to cache except image file existence(1.jpg, 1.webp?)
Moreover this does harm to kotlinx-io integration。

Reply:
We maybe also need to cache children of the root download location, but extract the caching logic out of UniFile SGTM.